### PR TITLE
Add SAFD paper.

### PR DESCRIPTION
**Pull request by czotti**
Review on "Scale Aware Face Detection".

### CODE REVIEW

File: deep-learning/_posts/2017-08-01-safd.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  title:   "Scale-Aware Face Detection"
+  venue:   "CVPR 2017"
+---
+
+This paper address a face detection problem and test their model on three face detection datasets, Face Detection Data Set and Benchmark ([FDDB](http://vis-www.cs.umass.edu/fddb/){:target="_blank"}), Multi-Attribute Labelled Faces ([MALF](http://www.cbsr.ia.ac.cn/faceevaluation/){:target="_blank"}) and Face Detection, Pose Estimation and Landmark Localization in the Wild ([AFW](http://www.ics.uci.edu/~xzhu/face/){:target="_blank"}).
+
+
+## Models
+
+They split the problem in two smaller one. They use a network to detect the scale of faces in the image and another one to place boundingbox on these faces.
```
Comment (jodoin) [inline deep-learning/_posts/2017-08-01-safd.md]: ones.

Comment (jodoin) [inline deep-learning/_posts/2017-08-01-safd.md]: a bounding box

File: deep-learning/_posts/2017-08-01-safd.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+### Scale Proposal Network (SPN)
+
+First, they use a CNN model (SPN) to predict an historgram of scales of faces in the images, as shown in Fig.3.
```
Comment (jodoin) [inline deep-learning/_posts/2017-08-01-safd.md]: in the image

File: deep-learning/_posts/2017-08-01-safd.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+     <img src="/deep-learning/images/safd/spn_refined.png">
+</div>
+
+### Region Proposal Network (RPN)
+
+Finally, after the proposals produced by the SPN, they rescale the image given the scaling histogram and predict the boundingbox face for all the rescaled images with the Single Scale RPN as in Fig.2.
+
+![](/deep-learning/images/safd/safd_pipeline.png){:align="middle"}
+
+The RPN is the Faster r-cnn model ([paper](https://arxiv.org/abs/1506.01497), [code](https://github.com/ShaoqingRen/faster_rcnn)). The inner network for these two models is made from [GoogleNet](https://arxiv.org/abs/1409.4842).
```
Comment (jodoin) [inline deep-learning/_posts/2017-08-01-safd.md]: They use Faster r-cnn as their RPN method.

File: deep-learning/_posts/2017-08-01-safd.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Training
+
+They generate the bounding box from keys points in faces to avoid the noise added when doing manual bounding box annotation. The keys points are left eye center, right eye center, nose, left mouth corner and right mouth corner. Each bounding box generated are square.
```
Comment (jodoin) [inline deep-learning/_posts/2017-08-01-safd.md]: key points in faces => facial landmarks

Comment (jodoin) [inline deep-learning/_posts/2017-08-01-safd.md]: The keys points are > The landmarks are

File: deep-learning/_posts/2017-08-01-safd.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+---
+layout: review
+title:  "Scale-Aware Face Detection"
+tags:   deep-learning CNN
```
Comment (jodoin) [inline deep-learning/_posts/2017-08-01-safd.md]: please add a "face detection" tag

File: deep-learning/_posts/2017-08-01-safd.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+pdf:   "https://arxiv.org/pdf/1706.09876.pdf"
+
+cite:
+
+  authors: "H.Zekun,L.Yu, Q.Hongwei, Y.Junjie, L.Xiu, H.Xiaolin"
+  title:   "Scale-Aware Face Detection"
+  venue:   "CVPR 2017"
+---
+
+This paper address a face detection problem and test their model on three face detection datasets, Face Detection Data Set and Benchmark ([FDDB](http://vis-www.cs.umass.edu/fddb/){:target="_blank"}), Multi-Attribute Labelled Faces ([MALF](http://www.cbsr.ia.ac.cn/faceevaluation/){:target="_blank"}) and Face Detection, Pose Estimation and Landmark Localization in the Wild ([AFW](http://www.ics.uci.edu/~xzhu/face/){:target="_blank"}).
```
Comment (Dref360) [inline deep-learning/_posts/2017-08-01-safd.md]: addresses

File: deep-learning/_posts/2017-08-01-safd.md (hunk 2 above)
Comment (Dref360) [inline deep-learning/_posts/2017-08-01-safd.md]: histogram

File: deep-learning/_posts/2017-08-01-safd.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+They refine the prediction of the SPN by smoothing it with a moving average and use the non max supression algorithm to have fewer scales as shown in Fig.4.
+
+<div align="middle">
+     <img src="/deep-learning/images/safd/spn_refined.png">
+</div>
+
+### Region Proposal Network (RPN)
+
+Finally, after the proposals produced by the SPN, they rescale the image given the scaling histogram and predict the boundingbox face for all the rescaled images with the Single Scale RPN as in Fig.2.
```
Comment (Dref360) [inline deep-learning/_posts/2017-08-01-safd.md]: phrasing : With the proposals...

**Comment by czotti** [comment]
Can I merge ?

File: deep-learning/_posts/2017-08-01-safd.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+## Training
+
+They generate the bounding box from keys points in faces to avoid the noise added when doing manual bounding box annotation. The keys points are left eye center, right eye center, nose, left mouth corner and right mouth corner. Each bounding box generated are square.
+
+For the scaling vector used in the SPN they use a gaussian function centered on the ground truth face size.
+
+## Results
+
+They report SOTA results on the three datasets with a **mean recal rate** of **88.15%** and an **average precision** of **98.77%**.
```
Comment (zhimingluo) [inline deep-learning/_posts/2017-08-01-safd.md]: mean recal rate -> mean recall

File: deep-learning/_posts/2017-08-01-safd.md (hunk 7 above)
Comment (zhimingluo) [inline deep-learning/_posts/2017-08-01-safd.md]: the SPN produce the scales not the proposals.

**Pull request merged**